include/rocksdb/table_properties.h

```diff
@@ -133,6 +133,9 @@ class TablePropertiesCollectorFactory {
  public:
   struct Context {
     uint32_t column_family_id;
+    // The level of the SST file (i.e, table), of which the properties are being
+    // collected. -1 if unknown or N/A
+    int file_level = -1;
```

Comment:
There's a reason that `level_at_creation` is called `level_at_creation` rather than `file_level`. Trivial moves can change the level of a file. Calling an immutable property `file_level` perpetuates a false idea that level is an immutable property of a file. It's very important that we not perpetuate this idea, or people see confusing behavior where things on one level behave as if they're on another level. By calling it `level_at_creation` we are making it clear that the only immutable level property of an SST file is its starting level. This should clue people into not assuming that an immutable level property reflects the current level of the file.

Reply:
Good point - I also agree with naming the variable precisely as the variable name conveys concepts and relations. Will fix this along with other suggestions above :)